src/iceberg/expression/literal.cc

```diff
@@ -0,0 +1,358 @@
+  bool rhs_is_nan = std::isnan(rhs);
+
+  // If both are NaN, check their signs
+  if (lhs_is_nan && rhs_is_nan) {
+    bool lhs_is_negative = std::signbit(lhs);
+    bool rhs_is_negative = std::signbit(rhs);
+
+    if (lhs_is_negative == rhs_is_negative) {
+      // Same sign NaN values are equivalent (no qNaN vs sNaN distinction)
+      return std::partial_ordering::equivalent;
+    }
+    // -NaN < NaN
+    return lhs_is_negative ? std::partial_ordering::less : std::partial_ordering::greater;
+  }
+
+  // For non-NaN values, use standard strong ordering
+  return std::strong_order(lhs, rhs);
+}
+
+// Three-way comparison operator
+std::partial_ordering Literal::operator<=>(const Literal& other) const {
+  // If types are different, comparison is unordered
+  if (type_->type_id() != other.type_->type_id()) {
+    return std::partial_ordering::unordered;
+  }
+
+  // If either value is AboveMax or BelowMin, comparison is unordered
+  if (IsAboveMax() || IsBelowMin() || other.IsAboveMax() || other.IsBelowMin()) {
+    return std::partial_ordering::unordered;
+  }
+
+  // Same type comparison for normal values
+  switch (type_->type_id()) {
+    case TypeId::kBoolean: {
+      auto this_val = std::get<bool>(value_);
+      auto other_val = std::get<bool>(other.value_);
+      if (this_val == other_val) return std::partial_ordering::equivalent;
+      return this_val ? std::partial_ordering::greater : std::partial_ordering::less;
+    }
+
+    case TypeId::kInt: {
+      auto this_val = std::get<int32_t>(value_);
+      auto other_val = std::get<int32_t>(other.value_);
+      return this_val <=> other_val;
+    }
+
+    case TypeId::kLong: {
+      auto this_val = std::get<int64_t>(value_);
+      auto other_val = std::get<int64_t>(other.value_);
+      return this_val <=> other_val;
+    }
+
+    case TypeId::kFloat: {
+      auto this_val = std::get<float>(value_);
+      auto other_val = std::get<float>(other.value_);
+      // Use strong_ordering for floating point as spec requests
+      return iceberg_float_compare(this_val, other_val);
+    }
+
+    case TypeId::kDouble: {
+      auto this_val = std::get<double>(value_);
+      auto other_val = std::get<double>(other.value_);
+      // Use strong_ordering for floating point as spec requests
+      return iceberg_float_compare(this_val, other_val);
+    }
+
+    case TypeId::kString: {
+      auto& this_val = std::get<std::string>(value_);
+      auto& other_val = std::get<std::string>(other.value_);
+      return this_val <=> other_val;
+    }
+
+    case TypeId::kBinary: {
+      auto& this_val = std::get<std::vector<uint8_t>>(value_);
+      auto& other_val = std::get<std::vector<uint8_t>>(other.value_);
+      return this_val <=> other_val;
+    }
+
+    default:
+      // For unsupported types, return unordered
+      return std::partial_ordering::unordered;
+  }
+}
+
+std::string Literal::ToString() const {
+  if (std::holds_alternative<BelowMin>(value_)) {
+    return "BelowMin";
```

Comment:
```suggestion
    return "belowMin";
```

To be consistent with java impl